source/communication_interfaces/src/sockets/UDPServer.cpp

```diff
@@ -0,0 +1,21 @@
+#include "communication_interfaces/sockets/UDPServer.hpp"
+
+namespace communication_interfaces::sockets {
+
+UDPServer::UDPServer() : UDPSocket(), client_address_() {}
+
+UDPServer::UDPServer(const std::list<std::shared_ptr<state_representation::ParameterInterface>>& parameters) :
+    UDPSocket(parameters), client_address_() {}
```

Comment:
Maybe I'm missing something but I don't understand where `client_address_` is updated, if at all. It seems to be a private property of the Server class, default initialised in the constructors and then used in the receive and send methods. What's going on there?

Reply:
Yeah you're observation is correct. However, this has been done like that in all examples I've seen (for example [here](https://gist.github.com/sunmeat/02b60c8a3eaef3b8a0fb3c249d8686fd) and [here](https://www.geeksforgeeks.org/udp-server-client-implementation-c/)). My interpretation is that the server doesn't need a client address because UDP is connectionless protocol. Would be different for TCP

Reply:
Oh, I see. Thanks for the explanation. I don't mind changing the signature of the receive function.